src/api/middleware.ts

```diff
@@ -104,9 +104,10 @@ export function allow(auth: IAuth): Function {
       const packageName = req.params.scope ? `@${req.params.scope}/${req.params.package}` : req.params.package;
       const packageVersion = req.params.filename ? getVersionFromTarball(req.params.filename) : undefined;
       const remote: RemoteUser = req.remote_user;
+      const tag = req.body['dist-tags'] ? Object.keys(req.body['dist-tags']).pop() : 'latest';
```

Comment:
Just one thing, I think it will make more sense to change `tag` -> `dist_tag` ?

Reply:
My opinion here is, hardcode `latest` is not a good idea, `latest` is the default tag but actually can be removed from any package manually. I've seen scenarios here where people uses other names by default and `latest` is missing completely.  I'm still checking this anyway.